### ISSUE TITLE
Custom sorting order using a chain based on two orders

### ISSUE THREAD

**Issue by ProgrammingLife** [ranger/ranger#2531]
#### Runtime Environment
- Operating system and version: Linux Mint 19.1
- Terminal emulator and version: gnome-terminal 3.28.1
- Python version: 3.6.8
- Ranger version/commit: ranger-master (built about 2 weeks ago)
- Locale: en_US.UTF-8

#### Current Behavior
It sorts either by name or by type. But if I press sort by natural and then press sort by type: (```o-n```, ```o-t```) - it sorts perfecty!

#### Expected Behavior
Sorting by name and grouping files by type/extension.

#### Possible Solutions
Setting not only one sorting order but a some sort of chain based on two sorting orders in ```rc.conf```, for instance: 
```
set sort natural,type
```

#### Steps to reproduce
1. Open a folder with files that have different types
2. Press: ```o-n```, ```o-t``` - I wanna exactly that sorting type setting in ```rc.conf```!

#### Traceback
n/a

**Comment by toonn** [comment]
Yeah, we want to do something similar to the filter_stack functionality. Maybe doing it more efficiently, rather than relying on sorts being stable and sorting the entire list again and again only falling back to a different sorting method whenever two items are considered equal for the main sorting method.

**Comment by cybardev** [comment]
Any way to achieve this automatically for now? Like by sorting by type *after* ranger is fully loaded? Just using the flag `--cmd="sort type"` doesn't seem to work.

**Comment by toonn** [comment]
@cybardev, you could try coming up with an `eval` command that does the two sorts. That should prevent the chain behavior where only the last sort is really applied.
